app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +25,35 @@ def show
     }
   end
 
+  def raw
+    @version ||= version_collection.find(params[:id])
+
+    expires_in 1.year, :public => true
+
+    if Archiver.external_archive_url?(@version.uri)
+      redirect_to @version.uri, status: 301 and return
+    elsif Archiver.store.contains_url?(@version.uri)
+      upstream = Archiver.store.get_file(@version.uri)
+
+      # Media type logic mostly cribbed from
+      # web-monitoring-db/app/jobs/analyze_change_job.rb
+      # Lines 176 to 180 in 658ae8c
+      # TODO: this will eventually be a proper field on `version`:
+      # https://github.com/edgi-govdata-archiving/web-monitoring-db/issues/199
+      meta = @version.source_metadata || {}
+      media = meta['media_type'] || meta['content_type'] || meta['mime_type']
+      if !media && meta['headers'].is_a?(Hash)
+        media = meta['headers']['content-type'] || meta['headers']['Content-Type']
+      elsif !media
+        media = upstream.content_type
+      end
+
+      send_data(upstream, type: media, disposition: 'inline')
+    else
+      raise ActiveRecord::RecordNotFound, "No raw content for #{@version.uuid}."
```

Comment:
I think it would be good not to co-opt an ActiveRecord error just to get a 404 here. If you want to raise, the `ApiError` module is probably what you should use. (We co-opt this error in `Change#find_by_api_id` only because that is doing an actual ActiveRecord lookup, and is meant to work and feel as if it is an ActiveRecord finder method from the outside.)